include/dlaf/factorization/qr/t_factor_mc.h

```diff
@@ -62,6 +90,157 @@ struct QR_Tfactor<Backend::MC, Device::CPU, T> {
                    common::Pipeline<comm::CommunicatorGrid>& serial_comm);
 };
 
+template <class T>
+hpx::future<matrix::Tile<T, Device::CPU>> gemvColumnT(
+    const bool& is_v0, const SizeType& k, hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_vi,
+    hpx::shared_future<common::internal::vector<T>>& taus,
+    hpx::future<matrix::Tile<T, Device::CPU>>& tile_t) {
+  auto gemv_func = hpx::util::unwrapping([=](const auto& vi, const auto& taus, auto&& t) {
+    DLAF_ASSERT(taus.size() == k, taus.size(), k);
+
+    for (SizeType j = 0; j < k; ++j) {
+      const T tau = taus[j];
+      // this is the x0 element of the reflector j
+      const TileElementIndex x0{j, j};
+
+      const TileElementIndex t_start{0, x0.col()};
+      const TileElementSize t_size{x0.row(), 1};
+
+      const SizeType first_element_in_tile = is_v0 ? x0.row() : 0;
+
+      // T(0:j, j) = -tau . V(j:, 0:j)* . V(j:, j)
+      // [j x 1] = [(n-j) x j]* . [(n-j) x 1]
+      TileElementSize va_size{vi.size().rows() - first_element_in_tile, x0.col()};
+      TileElementIndex va_start{first_element_in_tile, 0};
+      TileElementIndex vb_start{first_element_in_tile, x0.col()};
+
+      if (is_v0) {
+        t(x0) = tau;
+
+        // use implicit 1 for the 2nd operand
+        for (SizeType r = 0; !va_size.isEmpty() && r < va_size.cols(); ++r) {
+          const TileElementIndex i_v{va_start.row(), r + va_start.col()};
+          const TileElementIndex i_t{r + t_start.row(), t_start.col()};
+
+          DLAF_ASSERT_HEAVY(i_t.isIn(t.size()), i_t, t_size);
+          DLAF_ASSERT_HEAVY(i_v.isIn(vi.size()), i_v, vi.size());
+
+          t(i_t) = -tau * dlaf::conj(vi(i_v));
+        }
+
+        // skip alredy managed computations with implicit 1
+        va_start = va_start + TileElementSize{1, 0};
+        vb_start = vb_start + TileElementSize{1, 0};
+        va_size = {va_size.rows() - 1, va_size.cols()};
+      }
+
+      if (!va_size.isEmpty()) {
+        // clang-format off
+          blas::gemv(blas::Layout::ColMajor,
+              blas::Op::ConjTrans,
+              va_size.rows(), va_size.cols(),
+              -tau,
+              vi.ptr(va_start), vi.ld(),
+              vi.ptr(vb_start), 1,
+              1, t.ptr(t_start), 1);
+        // clang-format on
```

Comment:
Please note that if you disable automatic formatting to improve readability, it is your responsibility to check that the resulting formatting is consistent.

Please check ALL regions in which formatting is disabled.

Reply:
Formatting still wrong.